specs/networking/p2p-interface.md

```diff
@@ -0,0 +1,743 @@
+
+#### Mainnet
+
+[Noise Framework](http://www.noiseprotocol.org/) handshakes will be used for mainnet. libp2p Noise support [is in the process of being standardised](https://github.com/libp2p/specs/issues/195) in the libp2p project.
+
+Noise support will presumably include IX, IK and XX handshake patterns, and may rely on Curve25519 keys, ChaCha20 and Poly1305 ciphers, and SHA256 as a hash function. These aspects are being actively debated in the referenced issue [ETH 2.0 implementers are welcome to comment and contribute to the discussion.]
+
+## Protocol Negotiation
+
+#### Interop
+
+Connection-level and stream-level (see the rationale section below for explanations) protocol negotiation MUST be conducted using [multistream-select v1.0](https://github.com/multiformats/multistream-select/). Its protocol ID is: `/multistream/1.0.0`.
+
+#### Mainnet
+
+Clients MUST support [multistream-select 1.0](https://github.com/multiformats/multistream-select/) and MAY support [multiselect 2.0](https://github.com/libp2p/specs/pull/95). Depending on the number of clients that have implementations for multiselect 2.0 by mainnet, [multistream-select 1.0](https://github.com/multiformats/multistream-select/) may be phased out.
+
+## Multiplexing
+
+During connection bootstrapping, libp2p dynamically negotiates a mutually supported multiplexing method to conduct parallel conversations. This applies to transports that are natively incapable of multiplexing (e.g. TCP, WebSockets, WebRTC), and is omitted for capable transports (e.g. QUIC).
+
+Two multiplexers are commonplace in libp2p implementations: [mplex](https://github.com/libp2p/specs/tree/master/mplex) and [yamux](https://github.com/hashicorp/yamux/blob/master/spec.md). Their protocol IDs are, respectively: `/mplex/6.7.0` and `/yamux/1.0.0`.
+
+Clients MUST support [mplex](https://github.com/libp2p/specs/tree/master/mplex) and MAY support [yamux](https://github.com/hashicorp/yamux/blob/master/spec.md). If both are supported by the client, yamux must take precedence during negotiation. See the Rationale section of this document for tradeoffs.
+
+# ETH2 network interaction domains
+
+## Constants
+
+This section outlines constants that are used in this spec.
+
+- `RQRP_MAX_SIZE`: The max size of uncompressed req/resp messages that clients will allow.
+ Value: TBD
+- `GOSSIP_MAX_SIZE`: The max size of uncompressed gossip messages
+ Value: 1MB (estimated from expected largest uncompressed block size).
+- `SHARD_SUBNET_COUNT`: The number of shard subnets used in the gossipsub protocol.
+ Value: TBD
+
+## The gossip domain: gossipsub
+
+Clients MUST support the [gossipsub](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub) libp2p protocol.
+
+**Protocol ID:** `/meshsub/1.0.0`
+
+**Gossipsub Parameters**
+
+*Note: Parameters listed here are subject to a large-scale network feasibility study.*
+
+The following gossipsub parameters will be used:
+
+- `D` (topic stable mesh target count): 6
```

Comment:
Why is this called `D`? Other parameters have well formed names, can these have full names as well?

Reply:
I think `D` is the degree of the gossipsub overlay network.

Reply:
It was to be consistent with the specification. Didn't want to name it something different to the spec as it may add confusion. See here: https://github.com/libp2p/specs/tree/master/pubsub/gossipsub#meshsub-an-overlay-mesh-router
And here: https://github.com/libp2p/go-libp2p-pubsub/blob/master/gossipsub.go#L22

Reply:
Yeah. That said, I think the gossipsub spec could do a better job at variable naming. Especially now that it's being cross-referenced from the interwebs. I'll file an issue there.

Reply:
I'm sticking a link to that spec in there in the meantime